### ISSUE TITLE
Tree/TreeTable: Accessibility checkbox selection

### ISSUE THREAD

**Issue by kyybo** [primefaces/primereact#4691]
### Describe the bug

Hello, I hope you're well.
I'm here to report a small accessibility flaw in the Tree and TreeTable components.

As you mention in the 'Accessibility' section: "In checkbox selection, aria-checked is used instead of aria-selected."

When we create a new Tree with the 'checkbox' selection mode, we can see in the DOM that the checkbox elements created do indeed have an 'aria-checked="false"' attribute, which is perfect.
Then, if we select a tree node, the attribute becomes 'aria-checked="true"', so far so good.
BUT, when we deselect a node, the 'aria-checked' attribute disappears completely, whereas we'd like to have 'aria-checked="false"'.
The attribute reappears when the node is reselected, but disappears again when the node is deselected.

PS : The DataTable component with checkbox/radioButton selection mode doesn't seem to have this problem.

### PrimeReact version

9.6.0

### React version

18.x

### Language

TypeScript

### Build / Runtime

Create React App (CRA)

### Steps to reproduce the behavior

1. Create a simple Tree with checkbox selection mode
`<Tree value={nodes} selectionMode="checkbox" selectionKeys={selectedKeys} 
    onSelectionChange={(e) => setSelectedKeys(e.value)} className="w-full md:w-30rem" />`
2. In the DOM, we have :
`<div class="p-checkbox-box" role="checkbox" aria-checked="false" data-pc-section="checkbox"></div>`
which is good

3. Select a node and check the DOM :
`<div class="p-checkbox-box p-highlight" role="checkbox" data-pc-section="checkbox" aria-checked="true"></div>`
still good

4. Unselect the node and check the DOM :
`<div class="p-checkbox-box" role="checkbox" data-pc-section="checkbox"></div>`
The 'aria-checked' attribute is missing
